worlds/sc2/Client.py

```diff
@@ -966,7 +966,7 @@ def kerrigan_primal(ctx: SC2Context, kerrigan_level: int) -> bool:
         return kerrigan_level >= 35
     elif ctx.kerrigan_primal_status == KerriganPrimalStatus.option_half_completion:
         total_missions = len(ctx.mission_id_to_location_ids)
-        completed = len([(mission_id * VICTORY_MODULO + get_location_offset(mission_id)) in ctx.checked_locations
+        completed = sum([(mission_id * VICTORY_MODULO + get_location_offset(mission_id)) in ctx.checked_locations
                          for mission_id in ctx.mission_id_to_location_ids])
```

Comment:
```suggestion
        completed = sum((mission_id * VICTORY_MODULO + get_location_offset(mission_id)) in ctx.checked_locations
                         for mission_id in ctx.mission_id_to_location_ids)
```
Reduces memory use while we're at it, by turning it  into a generator.